sympy/physics/units/quantities.py

```diff
@@ -252,15 +252,18 @@ def free_symbols(self):
 def _Quantity_constructor_postprocessor_Add(expr):
     # Construction postprocessor for the addition,
     # checks for dimension mismatches of the addends, thus preventing
-    # expressions like `meter + second` to be created.
+    # expressions like `meter + second` to be created. For sake of
+    # core operations involving assumptions, the case of `unit + number`
+    # will be ignored.
 
```

Comment:
I still believe that `unit + number` should raise an exception.

Reply:
I think so as well.